packages/dds/shared-object-base/src/sharedObjectKernel.ts

```diff
@@ -0,0 +1,469 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { TypedEventEmitter } from "@fluid-internal/client-utils";
+import type { IFluidLoadable } from "@fluidframework/core-interfaces";
+import { assert, fail } from "@fluidframework/core-utils/internal";
+import {
+	IChannelStorageService,
+	type IChannel,
+	type IChannelAttributes,
+	type IChannelFactory,
+	type IChannelServices,
+	type IFluidDataStoreRuntime,
+} from "@fluidframework/datastore-definitions/internal";
+import type { IIdCompressor } from "@fluidframework/id-compressor/internal";
+import {
+	ISummaryTreeWithStats,
+	ITelemetryContext,
+	type IExperimentalIncrementalSummaryContext,
+	type IRuntimeMessageCollection,
+} from "@fluidframework/runtime-definitions/internal";
+import type { ITelemetryLoggerExt } from "@fluidframework/telemetry-utils/internal";
+
+import { IFluidSerializer } from "./serializer.js";
+import {
+	createSharedObjectKind,
+	SharedObject,
+	type ISharedObjectKind,
+	type SharedObjectKind,
+} from "./sharedObject.js";
+import { ISharedObjectEvents, type ISharedObject } from "./types.js";
+import type { IChannelView } from "./utils.js";
+
+/**
+ * Functionality specific to a particular kind of shared object.
+ * @remarks
+ * SharedObjects expose APIs for two consumers:
+ *
+ * 1. The runtime, which uses the SharedObject to summarize, load and apply ops.
+ * 2. The app, who uses the SharedObject to read and write data.
+ *
+ * There is some common functionality all shared objects use, provided by {@link SharedObject}.
```

Comment:
Nit: I generally recommend using the link in the first reference to a type (and using `\``s for subsequent references).